packages/commerce-sdk-react/src/hooks/ShopperSearch/query.ts

```diff
@@ -19,22 +25,59 @@ the product search hit. The search result contains only products that are online
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shoppersearch.shoppersearch-1.html#productsearch} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProductSearch = (
-    arg: Argument<Client['productSearch']>
-): UseQueryResult<DataType<Client['productSearch']>, Error> => {
-    const {shopperSearch: client} = useCommerceApi()
-    return useAsync(['product-search', arg], () => client.productSearch(arg))
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+): UseQueryResult<DataType<Client['productSearch']>, Error>
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: true},
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+): UseQueryResult<DataType<Client['productSearch']>, Error>
+function useProductSearch(
+    arg: UseProductSearchArg,
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+) {
```

Comment:
We need to define the return type here, which is going to be either product-search type OR the raw response.

```suggestion
): UseQueryResult<DataType<Client['productSearch']> | Response, Error> {
```

Reply:
FYI this is what's called the _implementation_ signature. When we do function overloading in Typescript, we first define the _overload_ signatures, which are then followed by the _implementation_ signature.

I like to think of the _implementation_ signature as a merge of all the above overload signatures. For more details, see: https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

Reply:
Got it. Thanks for pointing that out. Fixed the return type in the latest commit.